fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/LaoDetailViewModel.java

```diff
@@ -757,34 +714,24 @@ public void closeRollCall(int nextFragment) {
   }
 
   public void signMessage(WitnessMessage witnessMessage) {
-    String messageId = witnessMessage.getMessageId();
-    byte[] messageIdBuf = Base64.getUrlDecoder().decode(messageId);
-    /** Base 64 URL decoded message ID */
-    String signature;
-    /** Base 64 URL encoded signature of the message that we want to sign */
-    WitnessMessageSignature signatureMessage;
-    /** Message of type WitnessMessage that will be handled by LAORepository */
     Log.d(TAG, "signing message with ID " + witnessMessage.getMessageId());
     Lao lao = getCurrentLaoValue();
     if (lao == null) {
       Log.d(TAG, LAO_FAILURE_MESSAGE);
       return;
     }
+
     String channel = lao.getChannel();
 
     try {
-
-      KeysetHandle publicKeysetHandle = mKeysetManager.getKeysetHandle().getPublicKeysetHandle();
-      String publicKey = Keys.getEncodedKey(publicKeysetHandle);
-      byte[] sender = Base64.getUrlDecoder().decode(publicKey);
-      PublicKeySign signer = mKeysetManager.getKeysetHandle().getPrimitive(PublicKeySign.class);
-
+      KeyPair mainKey = mKeyManager.getMainKeyPair();
       // generate the signature of the message
-      signature = Signature.generateSignature(signer, messageIdBuf);
+      Signature signature = mainKey.sign(witnessMessage.getMessageId());
```

Comment:
Why not `generateSignature(...)` or a `getSignature(...)` ?

Reply:
I like the idea that a key signs some data and generate a signature.

But I am not very proficient with cryptographic terms. If you think that it is more meaningful to use `generateSignature`, then I can change it for sure.

Reply:
Let's keep it for now.